ldms/src/ldmsd/ldmsd_stream.c

```diff
@@ -253,6 +258,52 @@ static void event_cb(ldms_t x, ldms_xprt_event_t e, void *cb_arg)
 	sem_post(&conn_sem);
 }
 
+#define ROUND(len) ((((len)-1)|0XFFF)+1)
```

Comment:
If you scavenge code from ldmsd_stream_publish, you can send the file's contents a record at a time. This will avoid reading the whole file into memory.

Reply:
I refactored to code so that it reads the file in multiple chunks if the file is larger than max message length.